packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -4418,44 +4324,55 @@ class _ExpandedTextBoundary extends _TextBoundary {
 //
 // The innerTextBoundary must be a [_TextBoundary] that interprets the input
 // [TextPosition]s as code unit positions.
-class _CollapsedSelectionBoundary extends _TextBoundary {
+class _CollapsedSelectionBoundary extends TextBoundary {
```

Comment:
do we still need this, now that we always treat `TextPosition` as caret locations?

Reply:
Yes otherwise you won't be able to move to the next word if the current position is at the edge. This boundary move the position by one affinity. This may need a better name though...